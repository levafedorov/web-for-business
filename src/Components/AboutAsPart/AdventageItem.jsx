import React from 'react';
import PropTypes from 'prop-types';



export default function AdventageItem(props){
     

   const backgroundImg = {
       backgroundImage : `url(/photos/${props.name.trim().replace(' ', '-')}.jpg)`
   }
   
    return <div className="adventage-item">
           <div className="item-image" style={backgroundImg}>
           </div>
           <div className="adventage">
           {props.name}
           </div>
           </div>
           
}

AdventageItem.defaultProps = {
    name:"default name"
}

AdventageItem.propTypes = {
    name: PropTypes.string
}