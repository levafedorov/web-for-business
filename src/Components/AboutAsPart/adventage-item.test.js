import React from 'react';
import {shallow} from 'enzyme';
import AdventageItem from './AdventageItem';

describe("AdventageItem component", ()=>{
    const setUp = (props) => shallow(<AdventageItem {...props}/>);

    test("Check with props", ()=>{
       const component = setUp({name: 'check prop'})
       expect(component).toMatchSnapshot();    
    });
    test("Check with def props", ()=>{
        const component = setUp();
        expect(component).toMatchSnapshot();    
     });
});
