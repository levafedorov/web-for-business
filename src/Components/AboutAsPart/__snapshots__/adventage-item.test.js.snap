// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdventageItem component Check with def props 1`] = `
<div
  className="adventage-item"
>
  <div
    className="item-image"
    style={
      Object {
        "backgroundImage": "url(/photos/default-name.jpg)",
      }
    }
  />
  <div
    className="adventage"
  >
    default name
  </div>
</div>
`;

exports[`AdventageItem component Check with props 1`] = `
<div
  className="adventage-item"
>
  <div
    className="item-image"
    style={
      Object {
        "backgroundImage": "url(/photos/check-prop.jpg)",
      }
    }
  />
  <div
    className="adventage"
  >
    check prop
  </div>
</div>
`;
